qiskit/pulse/builder.py

```diff
@@ -0,0 +1,1996 @@
+
+        If ``lazy == True`` this circuit will extend a lazily constructed
+        quantum circuit. When an operation occurs that breaks the underlying
+        circuit scheduling assumptions such as adding a pulse instruction or
+        changing the alignment context the circuit will be
+        transpiled and scheduled into pulses with the current active settings.
+
+        Args:
+            circ: Circuit to call.
+            lazy: If false the circuit will be transpiled and pulse scheduled
+                immediately. Otherwise, it will extend the active lazy circuit
+                as defined above.
+        """
+        if lazy:
+            self._call_circuit(circ)
+        else:
+            self._compile_lazy_circuit()
+            self._call_circuit(circ)
+            self._compile_lazy_circuit()
+
+    @_requires_backend
+    def call_gate(self,
+                  gate: circuit.Gate,
+                  qubits: Tuple[int, ...],
+                  lazy: bool = True):
+        """Call the circuit ``gate`` in the pulse program.
+
+        The qubits are assumed to be defined on physical qubits.
+
+        If ``lazy == True`` this circuit will extend a lazily constructed
+        quantum circuit. When an operation occurs that breaks the underlying
+        circuit scheduling assumptions such as adding a pulse instruction or
+        changing the alignment context the circuit will be
+        transpiled and scheduled into pulses with the current active settings.
+
+        Args:
+            gate: Gate to call.
+            qubits: Qubits to call gate on.
+            lazy: If false the circuit will be transpiled and pulse scheduled
+                immediately. Otherwise, it will extend the active lazy circuit
+                as defined above.
+        """
+        try:
+            iter(qubits)
+        except TypeError:
+            qubits = (qubits,)
+
+        qc = circuit.QuantumCircuit(self.num_qubits)
+        qc.append(gate, qargs=qubits)
+        self.call_circuit(qc, lazy=lazy)
+
+
+def build(backend=None,
+          schedule: Optional[Schedule] = None,
+          default_alignment: str = 'left',
+          default_transpiler_settings: Optional[Dict[str, Any]] = None,
+          default_circuit_scheduler_settings: Optional[Dict[str, Any]] = None
+          ) -> ContextManager[Schedule]:
+    """A context manager for launching the imperative pulse builder DSL.
+
+    To enter a building context and starting building a pulse program:
+
+    .. code-block:: python
+        :emphasize-lines: 9
+
+        from qiskit import execute
+        from qiskit import pulse
+        from qiskit.test.mock import FakeOpenPulse2Q
+
+        backend = FakeOpenPulse2Q()
+
+        d0 = pulse.DriveChannel(0)
+
+        with pulse.build() as pulse_prog:
+            pulse.play(pulse.Constant(100,0.5), d0)
+
+        # execute on real backend
+        # qiskit.execute(pulse_prog, backend)
```

Comment:
Commented out code ❌ This could be wrapped up and explained in a follow up comment. This gives you the opportunity to clarify why you can't execute this program on `backend` when `backend` is `FakeOpenPulse2Q`.

Reply:
Fixed in 02b1c31c7.